---
title: "Lake Lugano 16S-rDNA ecology"
output: 
   html_document: 
      
      code_folding: show
      standalone: true
      self-contained: true
   pdf_document: default
   html_notebook: default
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
 echo = TRUE,
 include = TRUE,
 results = "show",
 cache = F,
 warning = FALSE,
 message = FALSE,
 fig.path = "figs/"
 # dev = c("pdf", "png")
 #dev.args = list(pdf = list(encoding = "WinAnsi", useDingbats = FALSE))
 )

require(tinytex)
```
&nbsp;

#### Reproducing the analysis published in: *Yuki Weber, Jaap S. Sinninghe Damst√©, Jakob Zopfi, Cindy De Jonge, Adrian Gili, Carsten J. Schubert, Fabio Lepori, Moritz F. Lehmann, Helge Niemann (2018): **"Redox-dependent niche differentiation of tetraether lipid producing bacteria: Evidence for multiple branched GDGT sources in lakes"***

&nbsp;

### Preparation

1.) Install 'R' and 'RStudio' on your machine.

2.) Download Supplementary Information Datasets 1 and 2 from the publisher's website and place files *dataset1.xls* and *dataset2.xls* into the working directory (i.e, the RStudio project folder).  

*<b>note:</b>*
To execute the following code in RStudio, run the respective code chunks (gray boxes) either by pressing *CMD + Shift + Enter* (Mac), or by clicking on the 'play' button in the upper right corner of rach code block.

&nbsp;

### Run the analysis

&nbsp;

#### Install phylo.lipids package
```{r}
# install devtools if not present and load
if (!"devtools" %in% rownames(installed.packages())) {
 install.packages("devtools")
}
require(devtools)

# install the phylo.lipids package from local source and load
if (!"phylo.lipids" %in% rownames(installed.packages())) {
 devtools::install_github("yukiweber/phylo.lipids", build_vignettes = F)
}
require(phylo.lipids)
```

&nbsp;

#### Load raw data files
```{r, echo=TRUE, include=TRUE, warning=T}
### load data files ----
## OTU table
lugano_otu =
  readxl::read_xls ("dataset1.xls", sheet = "otu.table" ) %>%
  as.data.frame()
# set otu names as rownames 
rownames(lugano_otu) = lugano_otu[, 1][] 
# remove name column
lugano_otu = lugano_otu[ ,-1]
  
## total DNA concentrations
lugano_sam_dat =
  readxl::read_xls ("dataset1.xls", sheet = "sample.data" ) %>%
  as.data.frame()
# set rownames as sample names (identical with otu table)
rownames(lugano_sam_dat) = colnames(lugano_otu)

## taxonomy table
lugano_tax =
  readxl::read_xls ("dataset1.xls", sheet = "tax.table" ) %>%
  as.data.frame()
# set otu names as rownames 
rownames(lugano_tax) = lugano_tax[,1][]
# remove name column
lugano_tax = lugano_tax[ ,-1]

## build phyloseq object ----
## contains OTU and taxonmoy tables, and sample data (here: DNA concentraions)
lugano_PHY =
  phyloseq::phyloseq (
    phyloseq::otu_table (as.matrix(lugano_otu), taxa_are_rows = TRUE),
    phyloseq::tax_table (as.matrix(lugano_tax)),
    phyloseq::sample_data (lugano_sam_dat)
  )
```

&nbsp;

#### Processing of DNA data
```{r, echo=TRUE, include=TRUE, warning=FALSE}
## Transform OTU data ----
lugano_PHY %>%
# remove taxa that have less than 40 reads across all samples
phylo.lipids::trim_taxa_sum (min = 40) %>% 
# remove taxa that occurr in less than 5 samples
phylo.lipids::trim_taxa_samp (min = 5, fraction = F) %>%
# normalize to total dna
phylo.lipids::norm_dna (.@sam_data$`yg.DNA.L-1`) %>%
# remove eucaryotes and archaea
phylo.lipids::remove_clades (., c("Chloroplast", "Mitochondria", "Archaea")) -> PHY1

PHY2 = 
  PHY1 %>%
  # remove 100 m sample because the corresponsding lipid sample was lost
  phyloseq::subset_samples (depth != 100)

# test if there are still eukaryotic sequences
if (class (try (phyloseq::subset_taxa(PHY2, Class=="Chloroplast"))) != "try-error" | 
    class (try (phyloseq::subset_taxa(PHY2, Family=="Mitochondria"))) != "try-error"
) { rm(PHY2)
    rep("ALARM: still containing Eukaryotes!!",6) }

## differentiate between classes of Proteobacteria ----
PHY2 %>%
 phylo.lipids::diff_proteo() -> PHY3
```

&nbsp;

#### Xxxxx
```{r, echo=TRUE, include=TRUE, warning=TRUE}
## x


plot_groups(phy = PHY3,
            label = "depth",
            k=22,
            otus = F,
            level = "Phylum",
            fancy = F,
            count = "avg"
            ) -> GG0

GG0$facet_pies
GG0$facet_groups
GG0$groups_pies_only
GG0$legend_only


```

&nbsp;

#### Xxxxxxxx
```{r lugano_cor_phy, fig.height=4}
# x


plot_groups(PHY3,
            label = "depth",
            k=22,
            otus = F,
            prune_groups = c("G5","G21","G3"),
            level = "Order",
            fancy = T,
            cutoff = 0.02,
            count = "avg"
            ) -> GG1

GG1$facet_pies
GG1$facet_groups
GG1$groups_pies_only
GG1$legend_only
GG1$groups_pies_full
```

