---
title: "Lake Lugano 16S-rDNA ecology"
author: "Yuki Weber"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
 echo = TRUE,
 include = TRUE,
 results = "show",
 cache = T,
 warning = FALSE,
 message = FALSE,
 fig.path = "figs/",
 fig.width = 8,
 fig.asp=1
 # dev = c("pdf", "png")
 #dev.args = list(pdf = list(encoding = "WinAnsi", useDingbats = FALSE))
 )

#require(tinytex)
```
&nbsp;
## Exploring bacterial diversity in ecological niches within the water column

&nbsp;
#### Install phylo.lipids package
```{r}
# install devtools if not present and load
if (!"devtools" %in% rownames(installed.packages())) {
 install.packages("devtools")
}
require(devtools)

# install the phylo.lipids development package and load 
if (!"phylo.lipids" %in% rownames(installed.packages())) {
 devtools::install_github("yukiweber/phylo.lipids", ref = "release/0.0.901", build_vignettes = T)
}
require(phylo.lipids)
```

&nbsp;

#### Load raw data files
(alternatively you can use your existing phyloseq objects)  
Be aware of the effects of using either relative abundance data or DNA concentrations 
(based on total DNA or qPCR etc.)

```{r, echo=TRUE, include=TRUE, warning=T}
### load data files ----
otu_names=
  readr::read_csv(
    system.file("extdata", "lugano.otu.table.csv", package = "phylo.lipids"),
    col_names = T ) %>% 
  as.data.frame() %>%
  dplyr::select(1) %>%
  as.vector() 
otu_names = otu_names[,1] 

otu =
  readr::read_csv(
    system.file("extdata", "lugano.otu.table.csv", package = "phylo.lipids"),
    col_names = T ) %>%
    as.data.frame() %>%
    dplyr::select(-1)
rownames(otu) = otu_names

sam_dat=
  readr::read_csv(
    system.file("extdata", "lugano.sample.data.csv", package = "phylo.lipids"),
    col_names = T ) %>% 
    as.data.frame()
rownames(sam_dat) = sam_dat$sample

tax = 
  readr::read_csv(
    system.file("extdata", "lugano.tax.table.csv", package = "phylo.lipids"),
    col_names = T ) %>%
    as.data.frame() %>%
    dplyr::select(-1)
rownames(tax) = otu_names

## build a phyloseq object ----
PHY= phyloseq::phyloseq (phyloseq::otu_table (as.matrix(otu), taxa_are_rows = TRUE),
                         phyloseq::tax_table (as.matrix(tax)),
                         phyloseq::sample_data (sam_dat)
                         )

#### Processing of DNA data

## Transform OTU data ----
PHY %>%
# remove taxa that have less than 40 reads across all samples
phylo.lipids::trim_taxa_sum (min = 50) %>% 
# remove taxa that occurr in less than 5 samples
phylo.lipids::trim_taxa_samp (min = 4, fraction = F) -> PHY01


PHY01 %>%
   remove_clades ( c("Chloroplast", "Mitochondria", "Archaea")) %>%
   norm_dna(relative = T) %>%
   diff_proteo() -> PHY_rel


# normalize to total dna
PHY01 %>%
phylo.lipids::norm_dna (dna = sam_dat$`Âµg.DNA.L-1`) %>%
# remove eucaryotes and archaea
phylo.lipids::remove_clades (., c("Chloroplast", "Mitochondria", "Archaea")) %>% 
## differentiate between classes of Proteobacteria  
phylo.lipids::diff_proteo() -> PHY3

# test if there are still eukaryotic sequences
if (class (try (phyloseq::subset_taxa(PHY3, Class=="Chloroplast"))) != "try-error" | 
    class (try (phyloseq::subset_taxa(PHY3, Family=="Mitochondria"))) != "try-error"
) { rm(PHY2)
    rep("ALARM: still containing Eukaryotes!!",6) }

```

&nbsp;

##### First explore the ecological niche distribution
```{r, echo=TRUE, include=TRUE, warning=TRUE}

## try to find 'k' values that give a good separation of the groups 
## pay attention to the red ribbons (2 SD error bounds)
## eventuallt set 'otus = TRUE'
## use the depth label to reverse scales and mek them numeric
## 'depth' has to be a column of phy@sam_data
## use 'count = "sum"' to use the actual reads/concentrations/relative abundance values
## if 'count = "n"', the NUMBER OF OTUS affiliated with each taxonomic group (e.g. Phylum) will be used,
## regardless of the abundance of the indivisual otus
 
P = PHY3

plot_groups(phy = P,
            label = "de",
            k = 25,
            #h=1.5,
            otus = T,
            level = "Phylum",
            fancy = F,
            count = "sum",
            ncol = 7,
            clust = "ward.D2",
            standardize = "sum", 
            key_col = 3
            #rev.scale = F
            ) -> GG0

# save the group list
GG0$group_dat -> g
```
```{r fig.asp=1, fig.width=8}
GG0$facet_groups
```

##### Then look at the phylogenetic diversity in each group. 

```{r, fig.asp=1}
GG0$facet_pies
```

Eventually adjust the cutoff value to reduce the number of taxonomic units.  
You exempt taxa of interest from the cutoff (i.e., always show them) and/or highlight taxa of interest.

```{r}

plot_groups(phy = P,
            use_groups = g, ############# use the groups saved previously
            label = "depth",
            otus = T,
            level = "Phylum",
            fancy = F,
            count = "sum",
            cutoff = 0.05, ################### 5% cutoff
            ncol = 7,
            key_col = 3,
            exempt = "Nitrospirae"
            ) -> GG1


plot_groups(phy = P,
            label = "depth",
            use_groups = g,
            otus = T,
            level = "Phylum",
            fancy = F,
            count = "sum",
            cutoff = 1, # 100% cutoff
            ncol = 7,
            key_col = 3,
            exempt = c("Nitrospirae","Gammaproteobacteria","Bacteroidetes"), ############ !
            highlight = c("Nitrospirae","Gammaproteobacteria", "Bacteroidetes") ############# !
            ) -> GG2


GG1$facet_pies
GG2$facet_pies


```

&nbsp;

Search for taxa of interest at lower taxonomic levels

```{r}
plot_groups(phy = P,
            label = "depth",
            use_groups = g,
            otus = T,
            level = "Family",
            fancy = F,
            count = "sum",
            cutoff = 1,
            ncol = 7,
            key_col = 3,
            exempt = c("Methylococcaceae","Crenotrichaceae","Nitrospirales Incertae Sedis"), ############ !
            highlight = c("Methylococcaceae","Crenotrichaceae","Nitrospirales Incertae Sedis") ############# !
            ) -> GG2

GG2$facet_pies
```

&nbsp;

Now let's have a closer look at the phylogenetic diversity in SELECTED cluster groups of interest

```{r, results='hide'}

## set fancy = TRUE to make overlay plots
## It's slow so do not use it on the full data set

Gs = paste0("G", c(
         5, 6, 13, 15
                     )) 
# G_sel = mapply( function(Gs) grep(Gs, names(g), value = T), Gs, SIMPLIFY = T)
G_sel = sapply( Gs, function(x) grep(x, names(g), value = T))

int_tax = c("Methylococcaceae","Crenotrichaceae","Nitrospirales Incertae Sedis")

plot_groups(phy = P,
            label = "depth",
            use_groups = g,
            otus = T,
            level = "Order",
            fancy = T,
            count = "sum",
            cutoff = 0.06,
            ncol = 7,
            key_col = 3,
            exempt = int_tax, ######## !
            prune_groups = G_sel, ####### !
            # highlight = int_tax,
            pie.scale = 1.2
            ) -> GG4
```
```{r, fig.asp=0.5, fig.width=10, fig.align='center'}

GG4$groups_pies_full
```



